### PR TITLE
Make _FileSink adhere to API spec.

### PR DESCRIPTION
**Pull request by tvolkert**

### CODE REVIEW

**Comment by tvolkert** [comment]
@johnmccutchan @cbracken @matanlurey

**Comment by johnmccutchan** [comment]
LGTM but maybe add a test?

File: lib/src/backends/memory/memory_file.dart [hunk 1]
```diff
@@ -264,23 +263,57 @@ class _FileSink implements io.IOSink {
...
+      onDone: finish,
+    );
+    return _streamCompleter.future;
+  }
 
   @override
-  Future flush() => new Future.value();
+  Future flush() {
+    _checkNotStreaming();
+    return new Future.value();
```
Comment (cbracken) [inline lib/src/backends/memory/memory_file.dart]: If you want to save a line, just make the block as async and kill the return.

**Comment by tvolkert** [comment]
Tests are forthcoming (I have a PR in progress that starts to add them)

**Pull request merged**